### PR TITLE
Add gatekeeper flag for publicly caching gallery pages

### PR DESCRIPTION
**Pull request by joshlory**

### CODE REVIEW

**Comment by philbogle** [comment]
LGTM

**Comment by philbogle** [comment]
Approved for checkin to staging.

**Comment by wjordan** [comment]
The change looks safe for the publicly-cached state, however I'm not sure that the session-state will function as expected when `public_caching_for_gallery` is set to `false`. Because the session cookie is whitelisted at the http-cache layer, Rails will never receive the session cookie for this path, so then if the client sends a POST request from this page it may receive an authenticity token failure.

**Comment by joshlory** [comment]
I've updated this PR to move `POST /gallery_save` to a different path.  @laurelfan @wjordan can you take a look?

**Comment by laurelfan** [comment]
@wjordan I already removed the authenticity token checking for the gallery_activity post for exactly this reason (we caught this in bug bashing last week).  see L8 in  dashboard/app/controllers/gallery_activities_controller.rb

**Comment by joshlory** [comment]
In that case I think 6ed7134d8c79db6ee8c055fbec22c519826aeb03 is still required, but bf02bd5cda337079eda255e71e10c684253eb5cc may not be?

File: dashboard/config/routes.rb [hunk 1]
```diff
@@ -12,12 +12,14 @@ def redirect_to_teacher_dashboard
 
   resources :user_levels, only: [:update]
 
-  resources :gallery_activities, path: '/gallery' do
+  resources :gallery_activities, only: [:index], path: '/gallery' do
```
Comment (wjordan) [inline dashboard/config/routes.rb:15]: do we need to move the `gallery_activities#destroy` action somewhere else as well, or is it currently unused?

Comment (joshlory) [inline dashboard/config/routes.rb:15]: I believe it's unused, but I didn't remove it to be extra conservative with this change.

**Comment by laurelfan** [comment]
There's a unit test that asserts that the save_to_gallery_url is /gallery -- I'd be surprised if that still passes?

I think if this change didn't require changing any tests then we need more tests or there's something wrong with our tests.

**Comment by wjordan** [comment]
:+1:

**Comment by philbogle** [comment]
Assume this PR is now defunct?

**Pull request closed without merging**